From 53320437e1a4403f6bd5b3bcc3d00318953107df Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Tue, 28 Feb 2012 15:40:56 +0900
Subject: [PATCH 02/97] rawparse: set a frame number when beginning playback

n_frames should be set calculated based on playback position
when the new segment event happens. Because it would be cleared by
the flush event if the seek event is generated.
---
 gst/rawparse/gstrawparse.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/gst/rawparse/gstrawparse.c b/gst/rawparse/gstrawparse.c
index cf0f5d0..e4520f5 100644
--- a/gst/rawparse/gstrawparse.c
+++ b/gst/rawparse/gstrawparse.c
@@ -663,6 +663,14 @@ gst_raw_parse_sink_event (GstPad * pad, GstEvent * event)
 
         ret = gst_pad_push_event (rp->srcpad, event);
       }
+
+      /* set n_frames calculated based on beginning time */
+      if (rate >= 0)
+        gst_raw_parse_convert (rp, GST_FORMAT_TIME, start, GST_FORMAT_DEFAULT,
+            (gint64 *) & rp->n_frames);
+      else
+        gst_raw_parse_convert (rp, GST_FORMAT_TIME, stop, GST_FORMAT_DEFAULT,
+            (gint64 *) & rp->n_frames);
       break;
     }
     default:
-- 
1.7.10.4

